client/src/pages/dashboard/components/Bank/FavoriteCurrency/index.js

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import { Card, TabContent } from "reactstrap";
+import axios from "axios";
+import { connect, useDispatch } from "react-redux";
+
+import Navigation from "./Navigation";
+import BasicCurrency from "./BasicCurrency";
+import SelectCurrencies from "./SelectCurrencies";
+import Date from "./Date";
+import { copyPartOfStr } from "../../../../../utils";
+
+import getCurrenciesData from "../../../../../store/currenciesData/actions";
+
+let options = [];
+
+const FavoriteCurrency = () => {
+  const dispatch = useDispatch();
+
+  const [currencies, setCurrencies] = useState([]);
+  const [basicCur, setBasicCur] = useState("");
+  const [selectCurrencies, setSelectCurrencies] = useState([]);
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [activeTab, setActiveTab] = useState("1");
+  const [save, setSave] = useState(false);
+
+  useEffect(() => {
+    axios.get("https://api.exchangeratesapi.io/latest").then((res) => {
+      setCurrencies(res.data.rates);
+    });
+
+    const curValues = selectCurrencies.map((cur) => cur.value).join();
+    if (basicCur && curValues.length && startDate && endDate) {
```

Comment:
Use instead of `curValues.length` size() method from lodash.

Reply:
https://github.com/dimapanasiuk/mern/commit/238d879b661c2520a0da63abaf15942813d3223b